src/game_api/script/lua_backend.cpp

```diff
@@ -177,6 +178,51 @@ CustomMovableBehavior* LuaBackend::make_custom_movable_behavior(std::string_view
     return custom_behavior.get();
 }
 
+sol::object LuaBackend::get_user_data(Entity& entity)
+{
+    if (user_datas.contains(entity.uid))
+    {
+        return user_datas[entity.uid];
+    }
+    return sol::nil;
+}
+sol::object LuaBackend::get_user_data(uint32_t uid)
+{
+    if (user_datas.contains(uid))
+    {
+        return user_datas[uid];
+    }
+    return sol::nil;
+}
+void LuaBackend::set_user_data(Entity& entity, sol::object user_data)
+{
+    if (!user_datas.contains(entity.uid))
+    {
+        entity.set_on_dtor(
```

Comment:
I think we need to store the returned hook id and clear the hook in `clear_all_callbacks` to make sure a reload is not gonna call stale code?

Reply:
Probably, I didn't address the fucked up cleanup yet